packages/console/src/pages/AuthStatusChecker/index.tsx

```diff
@@ -0,0 +1,279 @@
+          ? await requestStorageAccess()
+          : { granted: false, hasAccess: false };
+
+        // Check storage accessibility after potentially requesting access
+        const storageCheck = checkStorageAccess();
+
+        const debugInfo = {
+          isInIframe,
+          storageAccessible: storageCheck.accessible,
+          origin: currentOrigin,
+          hostname: currentHostname,
+        };
+
+        addDebugLog('info', 'Storage access check completed', {
+          storageCheck,
+          storageAccessResult,
+          debugInfo,
+        });
+
+        try {
+          if (!storageCheck.accessible) {
+            throw new Error(`localStorage not accessible: ${storageCheck.error}`);
+          }
+
+          const hasToken = isAuthenticated;
+
+          const response: AdminTokenStatusMessage = {
+            type: AuthMessageType.ADMIN_TOKEN_STATUS,
+            hasToken,
+            requestId,
+            debugInfo,
+          };
+
+          addDebugLog('sent', 'Sending success response', response);
+
+          // Send response back to the requesting origin
+          if (source && 'postMessage' in source) {
+            source.postMessage(response, { targetOrigin: event.origin });
+            addDebugLog('info', 'Response sent successfully');
+          } else {
+            addDebugLog('error', 'No valid source to send response to');
+          }
+        } catch (error) {
+          // Handle localStorage access errors (e.g., in private browsing mode, iframe restrictions)
+          const errorMessage = error instanceof Error ? error.message : 'Unknown error';
+
+          const errorResponse: AdminTokenErrorMessage = {
+            type: AuthMessageType.ADMIN_TOKEN_ERROR,
+            error: 'Failed to access localStorage',
+            requestId,
+            debugInfo: {
+              ...debugInfo,
+              errorDetails: errorMessage,
+            },
+          };
+
+          addDebugLog('sent', 'Sending error response', errorResponse);
+
+          if (source && 'postMessage' in source) {
+            source.postMessage(errorResponse, { targetOrigin: event.origin });
+            addDebugLog('info', 'Error response sent successfully');
+          } else {
+            addDebugLog('error', 'No valid source to send error response to');
+          }
+        }
+      } else {
+        addDebugLog('error', 'Invalid message format received', event.data);
+      }
+    };
+
+    // Add event listener for postMessage immediately
+    window.addEventListener('message', handleMessage);
+    addDebugLog('info', 'Message event listener added - ready to receive messages');
+
+    // Cleanup
+    return () => {
+      window.removeEventListener('message', handleMessage);
+      addDebugLog('info', 'Message event listener removed');
+    };
+  }, [addDebugLog, isAuthenticated]); // Empty dependency array to ensure this runs only once on mount
+
+  const clearDebugLogs = () => {
+    setDebugLogs([]);
+  };
```

Comment:
Functions passing to another component should be better wrapped inside a `useCallback` hook